mongo_repository_test.go

```diff
@@ -74,19 +77,47 @@ func TestMongoSave(t *testing.T) {
 		defer client.Disconnect(context.TODO())
 
 		publisher := NewEventPublisher()
-		repo := NewMongoRepository(database, &publisher)
+		repo, err := NewMongoRepository[*Student](database, &publisher)
+		assert.NoError(t, err)
 		object := Student{ID: uuid.NewString()}
 
 		object.SetGrade("a")
-		repo.Save(&object)
+		err = repo.Save(&object)
+		assert.NoError(t, err)
 		assert.Len(t, eventStreamFor(t, database, object.ObjectID()), 1)
 
 		object.SetGrade("a")
-		repo.Save(&object)
+		err = repo.Save(&object)
+		assert.NoError(t, err)
 		assert.Len(t, eventStreamFor(t, database, object.ObjectID()), 2)
-		repo.Save(&object)
+		err = repo.Save(&object)
+		assert.NoError(t, err)
 		assert.Len(t, eventStreamFor(t, database, object.ObjectID()), 2)
 	})
+
+	t.Run("Concurrency error", func(t *testing.T) {
+		client, database := connectTestMongo(t)
+		defer client.Disconnect(context.TODO())
+
+		publisher := NewEventPublisher()
+		repo, err := NewMongoRepository[*Student](database, &publisher)
+		assert.NoError(t, err)
+		object := Student{ID: uuid.NewString()}
+
+		object.SetGrade("a")
+		err = repo.Save(&object)
+		assert.NoError(t, err)
+
+		object2 := object
+		object2.SetGrade("b")
+		err = repo.Save(&object2)
+		assert.NoError(t, err)
+
+		object.SetGrade("c")
+		err = repo.Save(&object)
+		assert.Error(t, err)
+		assert.True(t, errors.Is(err, ConcurrencyError))
```

Comment:
Peut-être plus clair (+ simplification du dernier assert):

```suggestion
		objectCopy := object

		object.SetGrade("b")
		err = repo.Save(&object)
		assert.NoError(t, err)

		objectCopy.SetGrade("c")
		err = repo.Save(&objectCopy)
		assert.Error(t, err)
		assert.ErrorIs(t, err, ConcurrencyError)
```